src/grappler/ngraph_optimizer.cc

```diff
@@ -166,8 +189,42 @@ Status NgraphOptimizer::Optimize(tensorflow::grappler::Cluster* cluster,
     DumpGraphs(graph, idx, "unmarked", "Unmarked Graph");
   }
 
+  // Get backend + its configurations, to be attached to the nodes
+  // Precedence Order: RewriteConfig > Env Variable > BackendManager
+  string backend_name = BackendManager::GetCurrentlySetBackendName();
+  if (!config_backend_name.empty()) {
+    if (!BackendManager::IsSupportedBackend(backend_name)) {
+      return errors::Internal("NGRAPH_TF_BACKEND: ", config_backend_name,
+                              " is not supported");
+    }
+    backend_name = config_backend_name;
+    NGRAPH_VLOG(1) << "Setting backend from the RewriteConfig " << backend_name;
+  } else {
+    const char* ng_backend_env_value = std::getenv("NGRAPH_TF_BACKEND");
+    if (ng_backend_env_value != nullptr) {
+      string backend_env = std::string(ng_backend_env_value);
+      if (backend_env.empty() ||
+          !BackendManager::IsSupportedBackend(backend_env)) {
+        return errors::Internal("NGRAPH_TF_BACKEND: ", backend_env,
+                                " is not supported");
+      }
+      backend_name = backend_env;
+      NGRAPH_VLOG(1) << "Setting backend from the enviornment variable "
+                        "NGRAPH_TF_BACKEND = "
+                     << backend_name;
+    } else {
+      NGRAPH_VLOG(1) << "Setting backend from the BackendManager ";
+    }
+    // splits into {"ngraph_backend", "_ngraph_device_config"}
+    config_map = BackendManager::GetBackendAttributes(
+        backend_name);  // SplitBackendConfig
+    backend_name = config_map.at("ngraph_backend");
+    config_map.erase("ngraph_backend");
```

Comment:
Yes, we don't need this map entry for later. The ngraph_backend is sent to mark_for_clustering as the backend_name parameter where it is added as a node attr. The rest of the config options are sent to encapsulate_clusters as a map and that does not need ngraph_backend since it will already be an attr of the node.